### ISSUE TITLE
Asyncio task exception on the first PeerConnection with FastAPI

### ISSUE THREAD

**Issue by seysn** [aiortc/aiortc#460]
Hello,

I am trying to implement aiortc in a project that use FastAPI, so I tried to convert the example `server` and `webcam` with FastAPI and uvicorn. There's not so much difference, I kept the same logic, I can provide the python files if needed.

I have no problem with the base examples with aiohttp, but the problem I get with FastAPI + uvicorn is that the offer doesn't work on the first time but it works after that.

For example, here is the full traceback I get from the first call of the `server` example with FastAPI :
```
Traceback (most recent call last):
  File "/.../aiortc/venv/lib/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 396, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/.../aiortc/venv/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/.../aiortc/venv/lib/python3.8/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/.../aiortc/venv/lib/python3.8/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/.../aiortc/venv/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/.../aiortc/venv/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/.../aiortc/venv/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/.../aiortc/venv/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/.../aiortc/venv/lib/python3.8/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/.../aiortc/venv/lib/python3.8/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "/.../aiortc/venv/lib/python3.8/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/.../aiortc/venv/lib/python3.8/site-packages/fastapi/routing.py", line 201, in app
    raw_response = await run_endpoint_function(
  File "/.../aiortc/venv/lib/python3.8/site-packages/fastapi/routing.py", line 148, in run_endpoint_function
    return await dependant.call(**values)
  File "webcam.py", line 41, in offer
    await pc.setRemoteDescription(offer)
  File "/.../aiortc/venv/lib/python3.8/site-packages/aiortc/rtcpeerconnection.py", line 839, in setRemoteDescription
    await add_remote_candidates(iceTransport, media)
  File "/.../aiortc/venv/lib/python3.8/site-packages/aiortc/rtcpeerconnection.py", line 156, in add_remote_candidates
    await asyncio.gather(*coros)
  File "/.../aiortc/venv/lib/python3.8/site-packages/aiortc/rtcicetransport.py", line 285, in addRemoteCandidate
    await self._connection.add_remote_candidate(
  File "/.../aiortc/venv/lib/python3.8/site-packages/aioice/ice.py", line 354, in add_remote_candidate
    mdns_protocol = await get_or_create_mdns_protocol()
  File "/.../aiortc/venv/lib/python3.8/site-packages/aioice/ice.py", line 35, in get_or_create_mdns_protocol
    async with _mdns_lock:
  File "/usr/lib/python3.8/asyncio/locks.py", line 97, in __aenter__
    await self.acquire()
  File "/usr/lib/python3.8/asyncio/locks.py", line 203, in acquire
    await fut
RuntimeError: Task <Task pending name='Task-7' coro=<RTCIceTransport.addRemoteCandidate() running at /.../aiortc/venv/lib/python3.8/site-packages/aiortc/rtcicetransport.py:285> cb=[gather.<locals>._done_callback() at /usr/lib/python3.8/asyncio/tasks.py:758]> got Future <Future pending> attached to a different loop
```

What is even stranger is that "the first time" doesn't refer to the first request the server receive but is based of the browser. If I restart my server, the next peer connection will work flawlessly, but if I start another browser or open a tab in incognito mode without restarting the server, the next peer connection will fail and the server will print the traceback above.

I'm not familiar with asyncio and the WebRTC protocol so I really don't understand what happens and I can't say from which module the problem comes. I believe the problem comes from aiortc but it could be FastAPI, uvicorn or even my code, but I want to make sure by asking here before jumping into conclusions.

Also I had a similar problem when aiortc was on the version 1.0.0 (and aioice 0.6.18), but it didn't work at all and I had no exception on the server side, it was just stuck on checking state, so it seems that the issue evolved with the update of aiortc (or aioice).

```
OS: Ubuntu 20.04.1 LTS
Python: 3.8.5
Chromium: 88.0.4324.96
Packages version:
- aiortc==1.1.0
- aioice==0.7.2
- fastapi==0.63.0
- starlette==0.13.6
```

**Comment by seysn** [comment]
I think I solved the problem on my end by editing aioice.

I moved `get_or_create_mdns_protocol` as a method of `Connection`, also moved the lock as an attribute of that class. It shouldn't be a problem since it is not used elsewhere.

Maybe I'll do a pull request on the aioice project if it is seems to be a good solution.

**Comment by jlaine** [comment]
Yes, please move this issue to aioice. Be aware that what you are doing defeats the point of have a unique MDnsProtocol instance: avoiding redundant simultaneous requests for a given hostname, so it's probably not going to fly as-is.

Also, I'm puzzled by what this has to do with FastAPI?

**Comment by jlaine** [comment]
It would seem the issue is not so much with FastAPI as with uvloop, I am able to reproduce this with uvloop 0.14.0

**Comment by jlaine** [comment]
This is fixed in aioice 0.7.3 which is is on its way to PyPI.
